src/libraries/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/EdgeCasesTests.cs

```diff
@@ -536,6 +536,23 @@ public static void CryptographicAttributeObjectOidCtor()
             Assert.Equal(0, cao.Values.Count);
         }
 
+        [Fact]
+        public static void CryptographicAttributeObjectAsnOidNull()
+        {
+            var oid = new Oid(Oids.Sha1);
+#if NETCOREAPP
+            AssertExtensions.Throws<ArgumentException>("values", () =>
+            {
+                new CryptographicAttributeObject(oid, new AsnEncodedDataCollection(new AsnEncodedData(new byte[] { 1, 2, 3 })));
+            });
+#else
+            Assert.Throws<NullReferenceException>(() =>
+            {
+                new CryptographicAttributeObject(oid, new AsnEncodedDataCollection(new AsnEncodedData(new byte[] { 1, 2, 3 })));
+            });
+#endif
```

Comment:
I believe that we have a helper for this pattern:

```suggestion
            AssertExtensions.Throws<ArgumentException, NullReferenceException>(
                "values",
                () =>
                    new CryptographicAttributeObject(
                        oid,
                        new AsnEncodedDataCollection(new AsnEncodedData(new byte[] { 1, 2, 3 }))));
```

Reply:
Huh. Today I Learned. https://github.com/dotnet/runtime/blob/7ea26e68f9856df48fb6f48666fc643bbcbefbc6/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs#L138-L140

Reply:
That looks much cleaner, I updated the test to use this. I also moved the `AsnEncodedData` initialization into a variable to make it a bit more readable. I can put it back if need be though.